core/row/join.pl

```diff
@@ -8,22 +8,44 @@
   my ($leof, $reof) = (0, 0);
   my ($llimit, @lcols) = @$left_cols;
   my ($rlimit, @rcols) = @$right_cols;
-  while (!$leof && !$reof) {
-    chomp(my $lkey = join "\t", (split /\t/, my $lrow = <STDIN>, $llimit + 1)[@lcols]);
-    chomp(my $rkey = join "\t", (split /\t/, my $rrow = <$fh>,   $rlimit + 1)[@rcols]);
-    $reof ||= !defined $rrow;
-    $leof ||= !defined $lrow;
+  my @rrows = ();
+  my @lrows = ();
 
-    until ($lkey eq $rkey or $leof or $reof) {
-      chomp($rkey = join "\t", (split /\t/, $rrow = <$fh>, $llimit + 1)[@lcols]),
-        $reof ||= !defined $rrow until $reof or $rkey ge $lkey;
-      chomp($lkey = join "\t", (split /\t/, $lrow = <STDIN>, $rlimit + 1)[@rcols]),
-        $leof ||= !defined $lrow until $leof or $lkey ge $rkey;
-    }
+  chomp(my $lkey = join "\t", (split /\t/, my $lrow = <STDIN>, $llimit + 1)[@lcols]);
+  chomp(my $rkey = join "\t", (split /\t/, my $rrow = <$fh>,   $rlimit + 1)[@rcols]);
+
+  while (!$leof && !$reof) {
+    if ($lkey lt $rkey) {
+      print "$lkey < $rkey\n";
+      chomp($lkey = join "\t", (split /\t/, $lrow = <STDIN>, $llimit + 1)[@lcols]);
+      $leof ||= !defined $lrow;
+    } elsif ($lkey gt $rkey) {
+      print "$lkey > $rkey\n";
+      chomp($rkey = join "\t", (split /\t/, $rrow = <$fh>,   $rlimit + 1)[@rcols]);
+      $reof ||= !defined $rrow;
+    } else {
+      @rrows = ($rrow,);
```

Comment:
this will work - you can also say `@rrows = $rrow` and perl will do the right thing